runtime/autoload/provider/pythonx.vim

```diff
@@ -22,96 +24,122 @@ function! provider#pythonx#Require(host) abort
+        \ 'sys.exit(2*int(pkgutil.get_loader("%s") is None))',
         \ a:module)])
   return [v:shell_error, prog_version]
 endfunction
 
-" Returns array: [was_success, error_message]
-function! provider#pythonx#CheckForModule(prog, module, major_version) abort
-  let prog_path = exepath(a:prog)
-  if prog_path ==# ''
-    return [0, a:prog . ' not found in search path or not executable.']
-  endif
+function! s:satisfies_version(target, ref_major, ref_minimum) abort
+  return a:target =~# '^' . a:ref_major && luaeval('vim.cmp_versions(_A[1], _A[2]) >= 0', [a:target, a:ref_minimum])
+endfunction
 
-  let min_version = (a:major_version == 2) ? '2.6' : '3.3'
+" Returns array: [was_success, python_version, error_message]
+function! provider#pythonx#CheckForModule(prog_path, module, major_version) abort
+  let min_version = get(s:, 'min_python'.a:major_version.'_version')
 
   " Try to load module, and output Python version.
   " Exit codes:
   "   0  module can be loaded.
   "   2  module cannot be loaded.
   "   Otherwise something else went wrong (e.g. 1 or 127).
-  let [prog_exitcode, prog_version] = s:import_module(a:prog, a:module)
-
-  if prog_exitcode == 2 || prog_exitcode == 0
-    " Check version only for expected return codes.
-    if prog_version !~ '^' . a:major_version
-      return [0, prog_path . ' is Python ' . prog_version . ' and cannot provide Python '
-            \ . a:major_version . '.']
-    elseif prog_version =~ '^' . a:major_version && prog_version < min_version
-      return [0, prog_path . ' is Python ' . prog_version . ' and cannot provide Python >= '
-            \ . min_version . '.']
-    endif
+  let [prog_exitcode, prog_version] = s:import_module(a:prog_path, a:module)
+
+  " Check version only for expected return codes.
+  if (prog_exitcode == 2 || prog_exitcode == 0) && !s:satisfies_version(prog_version, a:major_version, min_version)
+    return [0, '', a:prog_path . ' is Python ' . prog_version . ' and cannot provide Python '
+          \ . a:major_version . ' >= ' . min_version . '.']
   endif
 
   if prog_exitcode == 2
-    return [0, prog_path.' does not have the "' . a:module . '" module. :help provider-python']
-  elseif prog_exitcode == 127
-    " This can happen with pyenv's shims.
-    return [0, prog_path . ' does not exist: ' . prog_version]
+    return [0, '', a:prog_path.' does not have the "' . a:module . '" module. :help provider-python']
   elseif prog_exitcode
-    return [0, 'Checking ' . prog_path . ' caused an unknown error. '
-          \ . '(' . prog_exitcode . ', output: ' . prog_version . ')'
-          \ . ' Report this at https://github.com/neovim/neovim']
+    return [0, '', 'Checking ' . a:prog_path . ' exited with error code ' . prog_exitcode . '. '
+          \ . 'If running the same executable from the command line works fine, report this at '
+          \ . "https://github.com/neovim/neovim. This was the output:\n\n" . prog_version . "\n"]
```

Comment:
This branch sounds like the user has a broken Python executable, I would encourage them to start by investigating whether they can even run it on the command line, before reporting an issue to Neovim.